src/resources/views/crud/fields/select2_from_ajax.blade.php

```diff
@@ -58,115 +57,152 @@
-                                    id: item[$connectedEntityKeyName]
+        <!-- select2_from_ajax field type js -->
+        @loadJsOnce('packages/select2/dist/js/select2.full.min.js')
+        @if (app()->getLocale() !== 'en')
+            @loadJsOnce('packages/select2/dist/js/i18n/' . app()->getLocale() . '.js')
+        @endif
+        @loadOnce('bpFieldInitSelect2FromAjaxElement')
+        <script>
+            if (!window.fetchSingleEntry) {
+                var fetchSingleEntry = function (element) {
+                    var $fetchUrl = element.attr('data-data-source');
+                    var $relatedAttribute = element.attr('data-field-attribute');
+                    var $relatedKeyName = element.attr('data-connected-entity-key-name');
+                    var $value = element.attr('data-value');
+                    var $return = {};
+
+                    return new Promise(function (resolve, reject) {
+                        $.ajax({
+                            url: $fetchUrl + '/single',
+                            data: {
+                                'q': $value
+                            },
+                            type: 'POST',
+                            success: function (result) {
+
+                                $key = result[$relatedKeyName];
+                                $value = result[$relatedAttribute];
+                                $pair = { [$relatedKeyName] : $key, [$relatedAttribute] : $value}
+                                $return = {...$return, ...$pair};
+
+                                $(element).attr('data-current-value', JSON.stringify($return));
+
+                                resolve(result);
+                            },
+                            error: function (result) {
+
+                                reject(result);
+                            }
+                        });
+                    });
+                };
+            }
+            function bpFieldInitSelect2FromAjaxElement(element) {
+                var form = element.closest('form');
+                var $placeholder = element.attr('data-placeholder');
+                var $minimumInputLength = element.attr('data-minimum-input-length');
+                var $dataSource = element.attr('data-data-source');
+                var $method = element.attr('data-method');
+                var $fieldAttribute = element.attr('data-field-attribute');
+                var $connectedEntityKeyName = element.attr('data-connected-entity-key-name');
+                var $includeAllFormFields = element.attr('data-include-all-form-fields')=='false' ? false : true;
+                var $allowClear = element.attr('data-column-nullable') == 'true' ? true : false;
+                var $dependencies = JSON.parse(element.attr('data-dependencies'));
+                var $ajaxDelay = element.attr('data-ajax-delay');
+
+
+                if (!$(element).hasClass("select2-hidden-accessible"))
```

Comment:
Please add comment about this - ```// do not initialise select2s that have already been initialised```. It's my fault for not adding this comment in the first place.